tests/core/data/test_mongo_data_node.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import copy
+import datetime
+import json
+from dataclasses import dataclass
+from enum import Enum
+from unittest import mock
+
+import mongomock
+import numpy as np
+import pandas as pd
+import pymongo
+import pytest
+
+from src.taipy.core.common.alias import DataNodeId
+from src.taipy.core.data.mongo import MongoDataNode
+from src.taipy.core.exceptions.exceptions import InvalidExposedType, MissingRequiredProperty
+from taipy.config.common.scope import Scope
+
+
+class MyCustomObject:
+    def __init__(self, foo=None, bar=None, *args, **kwargs):
+        self.foo = foo
+        self.bar = bar
+        self.args = args
+        self.kwargs = kwargs
+
+
+class AnotherCustomObject:
+    def __init__(self, id, integer, text):
+        self.id = id
+        self.integer = integer
+        self.text = text
+
+
+class MyCustomEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, AnotherCustomObject):
+            return {"__type__": "AnotherCustomObject", "id": o.id, "integer": o.integer, "text": o.text}
+        return super().default(self, o)
+
+
+class MyCustomDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kwargs):
+        super().__init__(object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, o):
+        if o.get("__type__") == "AnotherCustomObject":
+            return AnotherCustomObject(o["id"], o["integer"], o["text"])
+        else:
+            return o
+
+
+class TestMongoDataNode:
```

Comment:
Can you add tests that will test on actual MongoDB database? We can do a condition for such tests like saying if we detect a MongoDB database running, we will run those tests (tests on actual mongodb database), if not then we will skip them. You can refer to file `tests/core/data/test_sql_table_data_node.py` right after the import for the condition 😃

Reply:
As far as I can understand, the test cases for SQL only check for `pyodbc` package. The test are using mocking to test the database, not on actual SQL database.

Do you mean we'll skip the test by checking for actual MongoDB like this? Or removing mongomock?
```suggestion
try:
    pymongo.MongoClient(
        host=db_host,
        port=db_port,
        username=db_username,
        password=db_password,
        serverSelectionTimeoutMS = 2000
    )
    client.server_info()
except:
    pytest.skip("skipping tests because there is no MongoDB database.", allow_module_level=True)
    
class TestMongoDataNode:
```